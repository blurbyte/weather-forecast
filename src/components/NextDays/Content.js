import BaseContent from '../Content';

const Content = BaseContent.withComponent('section');

export default Content;
