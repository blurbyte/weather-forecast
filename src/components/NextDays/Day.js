import React from 'react';
import styled from 'styled-components';

import Text from '../Text';
import WeatherIcon from '../WeatherIcon';

const Wrapper = styled.li`
  margin-top: 2rem;
`;

const Info = styled.div`
  display: flex;
  margin-top: 0.4rem;
`;

const Temperatures = styled.div`
  margin-left: 2rem;
  margin-top: 0.2rem;
`;

const Temperature = styled(Text).attrs({
  big: true
})`
  line-height: 1.2;
`;

const Day = () => (
  <Wrapper>
    <Text big>Thursday</Text>
    <Info>
      <WeatherIcon small />
      <Temperatures>
        <Temperature>20</Temperature>
        <Temperature dimmed>14</Temperature>
      </Temperatures>
    </Info>
    <Text>Clear sky</Text>
  </Wrapper>
);

export default Day;
