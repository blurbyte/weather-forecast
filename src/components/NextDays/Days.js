import styled from 'styled-components';

const Days = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
`;

export default Days;
