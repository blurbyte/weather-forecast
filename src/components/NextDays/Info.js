import styled from 'styled-components';

import media from '../../styles/media';

const Info = styled.div`
  display: flex;
  margin: 0.4rem 0;

  ${media.tablet`
    margin-left: auto;
    width: calc(50% + 1.8rem);
  `};
`;

export default Info;
