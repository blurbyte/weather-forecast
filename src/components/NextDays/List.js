import styled from 'styled-components';

import media from '../../styles/media';

const List = styled.ul`
  display: grid;
  list-style: none;
  padding: 0;
  margin: 0;
  grid-template-columns: repeat(4, 1fr);
  border-top: 1px solid ${props => props.theme.primary};

  ${media.tablet`
    display: flex;
    flex-direction: column;
    padding-top: 1rem;
  `};
`;

export default List;
