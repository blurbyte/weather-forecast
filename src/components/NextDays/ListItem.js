import styled from 'styled-components';

import media from '../../styles/media';

const ListItem = styled.li`
  margin-top: 2rem;

  ${media.desktop`
    display: flex;
    flex-direction: column;
    align-items: center;
  `};

  ${media.tablet`
    margin-top: 0;
    flex-direction: row;
    padding: 0 2rem;
  `};
`;

export default ListItem;
