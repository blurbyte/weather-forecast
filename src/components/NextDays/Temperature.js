import styled from 'styled-components';

import media from '../../styles/media';
import Text from '../Text';

const Temperature = styled(Text).attrs({
  big: true
})`
  line-height: 1.2;

  &:first-child {
    margin-right: 1rem;
  }

  ${media.tablet`
    width: 3rem;
    text-align: right;
  `};
`;

export default Temperature;
