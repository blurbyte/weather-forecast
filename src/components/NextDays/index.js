import React from 'react';

import Content from '../Content';
import Days from './Days';
import Day from './Day';

const NextDays = () => (
  <Content>
    <Days>
      <Day />
      <Day />
      <Day />
      <Day />
    </Days>
  </Content>
);

export default NextDays;
