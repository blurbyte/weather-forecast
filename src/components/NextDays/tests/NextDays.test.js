import React from 'react';
import renderer from 'react-test-renderer';

import NextDays from '../index';

jest.mock('../../WeatherIcon', () => 'WeatherIcon');

test('renders correctly', () => {
  const component = renderer.create(<NextDays />);
  expect(component.toJSON()).toMatchSnapshot();
});
