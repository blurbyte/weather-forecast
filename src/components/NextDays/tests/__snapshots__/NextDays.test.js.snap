// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa hyUlrz"
>
  <ul
    className="sc-bwzfXH WOqUP"
  >
    <li
      className="sc-bxivhb gpoiXE"
    >
      <p
        className="sc-htpNat brfPLX"
      >
        Thursday
      </p>
      <div
        className="sc-ifAKCX iFqtqX"
      >
        <WeatherIcon
          small={true}
        />
        <div
          className="sc-EHOje cEIgwp"
        >
          <p
            className="sc-bZQynM guznUi sc-htpNat brfPLX"
          >
            20
          </p>
          <p
            className="sc-bZQynM guznUi sc-htpNat gZTtHB"
          >
            14
          </p>
        </div>
      </div>
      <p
        className="sc-htpNat gyyIpD"
      >
        Clear sky
      </p>
    </li>
    <li
      className="sc-bxivhb gpoiXE"
    >
      <p
        className="sc-htpNat brfPLX"
      >
        Thursday
      </p>
      <div
        className="sc-ifAKCX iFqtqX"
      >
        <WeatherIcon
          small={true}
        />
        <div
          className="sc-EHOje cEIgwp"
        >
          <p
            className="sc-bZQynM guznUi sc-htpNat brfPLX"
          >
            20
          </p>
          <p
            className="sc-bZQynM guznUi sc-htpNat gZTtHB"
          >
            14
          </p>
        </div>
      </div>
      <p
        className="sc-htpNat gyyIpD"
      >
        Clear sky
      </p>
    </li>
    <li
      className="sc-bxivhb gpoiXE"
    >
      <p
        className="sc-htpNat brfPLX"
      >
        Thursday
      </p>
      <div
        className="sc-ifAKCX iFqtqX"
      >
        <WeatherIcon
          small={true}
        />
        <div
          className="sc-EHOje cEIgwp"
        >
          <p
            className="sc-bZQynM guznUi sc-htpNat brfPLX"
          >
            20
          </p>
          <p
            className="sc-bZQynM guznUi sc-htpNat gZTtHB"
          >
            14
          </p>
        </div>
      </div>
      <p
        className="sc-htpNat gyyIpD"
      >
        Clear sky
      </p>
    </li>
    <li
      className="sc-bxivhb gpoiXE"
    >
      <p
        className="sc-htpNat brfPLX"
      >
        Thursday
      </p>
      <div
        className="sc-ifAKCX iFqtqX"
      >
        <WeatherIcon
          small={true}
        />
        <div
          className="sc-EHOje cEIgwp"
        >
          <p
            className="sc-bZQynM guznUi sc-htpNat brfPLX"
          >
            20
          </p>
          <p
            className="sc-bZQynM guznUi sc-htpNat gZTtHB"
          >
            14
          </p>
        </div>
      </div>
      <p
        className="sc-htpNat gyyIpD"
      >
        Clear sky
      </p>
    </li>
  </ul>
</div>
`;
